### ISSUE TITLE
Error indexing _users db

### ISSUE THREAD

**Issue by callahanchris** [rnewson/couchdb-lucene#219]
I'm trying to implement an indexed search over all users in the `_users` database on couchdb using couchdb-lucene and am getting the following error: `{"reason":"error_creating_index","code":500}`. Not sure if it is possible to index the `_users` database but maybe I am doing something wrong!

Here's the original request:
```
curl http://admin:admin@127.0.0.1:5984/_fti/local/_users/_design/users/by_name?q=chris
```

and the response:
```
{"reason":"error_creating_index","code":500}
```

Here's the output I'm getting from couchdb-lucene:
```
2015-05-19 13:07:14,308 WARN [_users] Exiting after init() raised exception.
org.apache.http.client.HttpResponseException: {"error":"forbidden","reason":"Only admins can access _all_docs of system databases."}

	at com.github.rnewson.couchdb.lucene.util.ErrorPreservingResponseHandler.handleResponse(ErrorPreservingResponseHandler.java:36)
	at com.github.rnewson.couchdb.lucene.util.ErrorPreservingResponseHandler.handleResponse(ErrorPreservingResponseHandler.java:28)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:735)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:709)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:700)
	at com.github.rnewson.couchdb.lucene.couchdb.HttpUtils.execute(HttpUtils.java:36)
	at com.github.rnewson.couchdb.lucene.couchdb.HttpUtils.get(HttpUtils.java:40)
	at com.github.rnewson.couchdb.lucene.couchdb.Database.getAllDesignDocuments(Database.java:56)
	at com.github.rnewson.couchdb.lucene.DatabaseIndexer.init(DatabaseIndexer.java:777)
	at com.github.rnewson.couchdb.lucene.DatabaseIndexer.run(DatabaseIndexer.java:438)
	at java.lang.Thread.run(Thread.java:745)
```

Here's the couchdb log:
```
[info] [<0.113.0>] 127.0.0.1 - - GET /_users/ 200
[info] [<0.113.0>] 127.0.0.1 - - GET /_users/_local%2Flucene 200
[info] [<0.113.0>] 127.0.0.1 - - GET /_users/ 200
[info] [<0.113.0>] 127.0.0.1 - - GET /_users/_all_docs?startkey=%22_design%22&endkey=%22_design0%22&include_docs=true 403
[info] [<0.106.0>] 127.0.0.1 - - GET /_fti/local/_users/_design/users/by_name?q=chris 500
```

My ddoc:
```
{
  "_id": "_design/users",
  "fulltext": {
    "by_name": {
      "index": function(doc) {
        var ret = new Document();
        ret.add(doc.name);
        return ret;
      }
    }
  }
}
```

and here's my `couchdb-lucene.ini`:
```
[httpd_global_handlers]
_fti = {couch_httpd_proxy, handle_proxy_req, <<"http://127.0.0.1:5985">>}

[local]
url = http://admin:admin@127.0.0.1:5984/
```

In the couchdb config, `require_valid_user` is set to `false`.

On other dbs, I'm getting normal output from couchdb-lucene.

Any help is appreciated, thanks!

**Comment by leerlemoniii** [comment]
did you ever find solution for this.  I am about to do something similar, and wanted to see what was possible

**Comment by callahanchris** [comment]
@leerlemoniii Unfortunately I haven't found a solution to this exact problem so far. I ended up keeping a minimal `_users` database and moved most of the user metadata into a separate database. I had no problems using couchdb-lucene to index this other database.

**Comment by rnewson** [comment]
Ah, sorry, I missed this. The message is clear, though. You need admin rights to read the whole _users db, so check that couchdb-lucene.ini is logging in to couchdb with admin username and password. This is a CouchDB security restriction.

**Comment by callahanchris** [comment]
Thanks for getting back on this. Turns out I had updated the wrong `couchdb-lucene.ini` file. Previously I had put the admin username and password in the `couchdb-lucene.ini` installed by [homebrew](https://github.com/Homebrew/homebrew/blob/b2e847dfe56637ebd0581b91e01e3b0733646607/Library/Formula/couchdb-lucene.rb#L43) (located at `/usr/local/etc/couchdb/local.d/couchdb-lucene.ini` on my machine), which did not seem to get picked up by couchdb-lucene.

Instead, I just updated the `couchdb-lucene.ini` in the package installed by homebrew (located at `/usr/local/Cellar/couchdb-lucene/1.0.2/libexec/conf/couchdb-lucene.ini` on my machine). Works like a charm now!
